### DISCUSSION TITLE
npm ERR! Cannot destructure property 'name' of 'node' as it is undefined

### DISCUSSION THREAD

**Discussion by genesis16** [gatsbyjs/gatsby#33676] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Hello  when running Gatsby develop, I'm getting notified that Gatsby image plugin can not be found, perhaps you need to install it. And when I do install it, I'm getting an NPM error saying cannot destructure property of node as it is undefined. I have added gatsby image plugin to my gatsby-config.js file, it was working yesterday it just popped up today. 

ERROR 

UNHANDLED REJECTION Unable to find plugin "gatsby-plugin-image". Perhaps you need to install its package?

  Error: Unable to find plugin "gatsby-plugin-image". Perhaps you need to install its package?
  
  - load.js:105 resolvePlugin
    [jjgatsby]/[gatsby]/src/bootstrap/load-plugins/load.js:105:11
  
  - load.js:120 processPlugin
    [jjgatsby]/[gatsby]/src/bootstrap/load-plugins/load.js:120:20
  
  - load.js:181 forEach
    [jjgatsby]/[gatsby]/src/bootstrap/load-plugins/load.js:181:20
  
  - Array.forEach
  
  - load.js:180 loadPlugins
    [jjgatsby]/[gatsby]/src/bootstrap/load-plugins/load.js:180:20
  
  - index.js:42 loadPlugins
    [jjgatsby]/[gatsby]/src/bootstrap/load-plugins/index.js:42:19
  
  - index.js:105 module.exports
    [jjgatsby]/[gatsby]/src/bootstrap/index.js:105:34
  
  - task_queues.js:95 processTicksAndRejections
    internal/process/task_queues.js:95:5
  
  - develop.js:343 module.exports
    [jjgatsby]/[gatsby]/src/commands/develop.js:343:29
  

not finished load plugins - 0.384s

janejames@Jane-James jjgatsby % npm install gatsby-plugin-image gatsby-plugin-sharp gatsby-source-filesystem gatsby-transformer-sharp
npm ERR! code ERESOLVE
npm ERR! Cannot destructure property 'name' of 'node' as it is undefined.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/janejames/.npm/_logs/2021-10-26T03_28_39_026Z-debug.log

### Reproduction Link

https://github.com/genesis16/jjgatsby

### Steps to Reproduce

1. Run gatsby develop
2. npm install gatsby-plugin-image gatsby-plugin-sharp gatsby-source-filesystem gatsby-transformer-sharp
3. NPM error
...

### Expected Result

Gatsby to load my site on localhost 

### Actual Result

NPM error, name of node is undefined

### Environment

```shell
System:
    OS: macOS 11.6
    CPU: (4) x64 Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.17.6 - /usr/local/bin/node
    npm: 8.1.1 - /usr/local/bin/npm
  Languages:
    Python: 2.7.15 - /usr/local/bin/python
  Browsers:
    Chrome: 95.0.4638.54
    Edge: 94.0.992.31
    Safari: 14.1.2
  npmPackages:
    gatsby: ^2.10.0 => 2.10.0
    gatsby-plugin-image: ^0.7.2 => 0.7.2
    gatsby-plugin-manifest: ^2.12.1 => 2.12.1
    gatsby-plugin-offline: ^3.10.2 => 3.10.2
    gatsby-plugin-react-helmet: ^3.10.0 => 3.10.0
    gatsby-plugin-react-svg: ^3.0.1 => 3.0.1
    gatsby-plugin-sharp: ^2.14.4 => 2.14.4
    gatsby-plugin-stylus: ^2.10.0 => 2.10.0
    gatsby-remark-images: ^3.11.1 => 3.11.1
    gatsby-remark-reading-time: ^1.1.0 => 1.1.0
    gatsby-source-filesystem: ^2.11.1 => 2.11.1
    gatsby-transformer-remark: ^4.11.0 => 4.11.0
    gatsby-transformer-sharp: ^2.12.1 => 2.12.1
  npmGlobalPackages:
    gatsby-cli: 3.14.1
```

### Config Flags

no

**Comment by LekoArts** [comment]
Hi!

I'm not sure if your reproduction steps were exactly what you did, but installing npm dependencies while `gatsby develop` is running won't work. The output at "Environment" also shows some pretty outdated versions but looking at your repository it should be on Gatsby v3.

Maybe npm screwed the dependency tree up at some point, so you could do the following:

- Delete the `.cache` and `public` folder
- Delete the `node_modules` folder
- Delete `package-lock.json`
- Run `npm install --legacy-peer-deps`
